src/main/java/org/terasology/launcher/util/FileUtils.java

```diff
@@ -147,18 +147,25 @@ public static void copyFolder(final Path source, final Path destination) throws
     }
 
     /**
-     * Ensures that a directory exists and is empty.
+     * Ensures that a directory exists, is a directory and is empty.
      *
      * Deletes directory contents if the directory is not empty. If the directory does not exist, it is created.
      * The directory itself is not deleted.
      *
      * @param directory absolute path to the directory
+     *
+     * @throws IOException The path is not a directory
      */
     public static void ensureEmptyDir(final Path directory) throws IOException {
-        if (Files.notExists(directory)) {
+        if (!Files.exists(directory)) {
             Files.createDirectories(directory);
         } else {
-            deleteDirectoryContent(directory);
+
+            if (!Files.isDirectory(directory)) {
+                throw new IOException("Directory is not a directory! " + directory);
+            } else {
```

Comment:
I'm not entirely sure, if we want to put that here.
`deleteDirectoryContent` will throw a `NotDirectoryException` on calling `Files.list(directory)` if `directory` is not a directory. While this should be a well-understandable exception, I think it makes sense to check before-hand.
I put the check into `ensureEmptyDir` to align it with `ensureWritableDir` which includes this check as well, but maybe it would be better off in `deleteDirectoryContent`...
@praj-foss @skaldarnar what do you think?

Reply:
This needs a revisit anyways. Some notes on the current state and where it might be headed:
- `Files.createDirectories(path)` will create directory (if possible) or return the directory if it exists. If it is no directory, it will throw an exception - pretty much what we are trying to do as a first step here and in `ensureWritableDir`. Maybe we can make better use of it.
- I'm not really happy with the `void` return types and the exceptional behavior. Maybe we can hide that and instead expose `Optional<Path>` or the like for internal usage (as abstraction over the underlying file system operations).
- Seems like we are building this up in multiple steps: (1) ensure directory exists, (2) ensure directory is writable, (3) ensure directory is empty, ... Not sure how we could build a nice functional interface for that, or whether it is actually worth it 🙄 
- ... and file attributes: https://docs.oracle.com/javase/8/docs/api/java/nio/file/attribute/package-summary.html 🤷‍♂